### PR TITLE
Suppress or fix all current FxCop issues

### PR DESCRIPTION
**Pull request by spaceisfun**

### CODE REVIEW

File: src/oss/Pe-Utility/Pe-Utility.csproj [hunk 1]
```diff
@@ -5,4 +5,11 @@
     <RootNamespace>Pe_Utility</RootNamespace>
   </PropertyGroup>
 
+  <!-- Added for Factory Orchestrator -->
+  <PropertyGroup>
+    <RunAnalyzersDuringBuild>false</RunAnalyzersDuringBuild>
```
Comment (hajya) [inline src/oss/Pe-Utility/Pe-Utility.csproj:10]: impact?

File: src/oss/IpcFramework/JKang.IpcServiceFramework.TcpTests/JKang.IpcServiceFramework.TcpTests.csproj [hunk 2]
```diff
@@ -21,6 +21,12 @@
   </ItemGroup>
 
   <!-- Added for Factory Orchestrator -->
+  <PropertyGroup>
+    <RunAnalyzersDuringBuild>false</RunAnalyzersDuringBuild>
```
Comment (hajya) [inline src/oss/IpcFramework/JKang.IpcServiceFramework.TcpTests/JKang.IpcServiceFramework.TcpTests.csproj:25]: impact?

**Review by hajya — COMMENTED** [review]
my overall feedback here is that while i can see / review this for mechanically ensuring you are correctly 'addressing' (fix or suppress), it's hard to tell what the impact is for the ones being suppressed and/or why its okay.

are there norms / standards for how this is communicated/documented?

it's possible it's my lack of familiariaty and approach to always wanting to understand 'why' some small changes are being made that leads me to think this is bigger / larger than a normal convention but it would be helpful to somehow educate on why these are needed/okay.

File: src/oss/Pe-Utility/Pe-Utility.csproj (hunk 1 above)
Comment (spaceisfun) [reply src/oss/Pe-Utility/Pe-Utility.csproj:10]: Disables FxCop on this project

File: src/oss/IpcFramework/JKang.IpcServiceFramework.TcpTests/JKang.IpcServiceFramework.TcpTests.csproj (hunk 2 above)
Comment (spaceisfun) [reply src/oss/IpcFramework/JKang.IpcServiceFramework.TcpTests/JKang.IpcServiceFramework.TcpTests.csproj:25]: Disables FxCop on this project

**Comment by spaceisfun** [comment]
> my overall feedback here is that while i can see / review this for mechanically ensuring you are correctly 'addressing' (fix or suppress), it's hard to tell what the impact is for the ones being suppressed and/or why its okay.
> 
> are there norms / standards for how this is communicated/documented?
> 
> it's possible it's my lack of familiariaty and approach to always wanting to understand 'why' some small changes are being made that leads me to think this is bigger / larger than a normal convention but it would be helpful to somehow educate on why these are needed/okay.

@hajya generally my thoughts are:
- I don't care about Warnings in test code
- I fix most fixible FxCop Warning in code we wrote. Not every Warning is accurate/fixable.
- I fix Warnings in forked 3rd party OSS only if they seem likely to cause performance or reliability issues. I try to keep 3rd party OSS as true to the OSS source as possible.
Here are the Warnings from our latest CI build:

	Line 2480: 2020-11-19T02:05:18.4186261Z ##[warning]src\oss\IpcFramework\JKang.IpcServiceFramework.Core\IpcFaultException.cs(8,18): Warning CA1032: Add the following constructor to IpcFaultException: public IpcFaultException().
	Line 2483: 2020-11-19T02:05:18.4219225Z ##[warning]src\oss\IpcFramework\JKang.IpcServiceFramework.Core\IpcFaultException.cs(8,18): Warning CA1032: Add the following constructor to IpcFaultException: public IpcFaultException(string message).
	Line 2486: 2020-11-19T02:05:18.4300478Z ##[warning]src\oss\IpcFramework\JKang.IpcServiceFramework.Core\IpcFaultException.cs(8,18): Warning CA1032: Add the following constructor to IpcFaultException: public IpcFaultException(string message, Exception innerException).
	Line 2511: 2020-11-19T02:05:19.6022714Z ##[warning]src\oss\IpcFramework\JKang.IpcServiceFramework.Client.Tcp\TcpIpcClient.cs(14,22): Warning CS0169: The field 'TcpIpcClient<TInterface>._isDisposed' is never used
	Line 2514: 2020-11-19T02:05:19.6033855Z ##[warning]src\oss\IpcFramework\JKang.IpcServiceFramework.Client.Tcp\TcpIpcClient.cs(14,22): Warning CA1823: Unused field '_isDisposed'.
	Line 3023: 2020-11-19T02:05:36.2291078Z ##[warning]src\PowerShellLibrary\CmdletClasses.cs(11,17): Warning CS1591: Missing XML comment for publicly visible type or member 'GuidTypes'
	Line 3026: 2020-11-19T02:05:36.2307360Z ##[warning]src\PowerShellLibrary\CmdletClasses.cs(13,9): Warning CS1591: Missing XML comment for publicly visible type or member 'GuidTypes.Task'
	Line 3029: 2020-11-19T02:05:36.2320974Z ##[warning]src\PowerShellLibrary\CmdletClasses.cs(14,9): Warning CS1591: Missing XML comment for publicly visible type or member 'GuidTypes.TaskList'
	Line 3032: 2020-11-19T02:05:36.2334491Z ##[warning]src\PowerShellLibrary\CmdletClasses.cs(15,9): Warning CS1591: Missing XML comment for publicly visible type or member 'GuidTypes.TaskRun'
	Line 3035: 2020-11-19T02:05:36.2360763Z ##[warning]src\PowerShellLibrary\CmdletClasses.cs(11,17): Warning CA1717: Only FlagsAttribute enums should have plural names
	**Line 3094: 2020-11-19T02:05:38.2842233Z ##[warning]src\oss\IpcFramework\JKang.IpcServiceFramework.Hosting.NamedPipe\NamedPipeNative.cs(14,27): Warning CA1060: Move pinvokes to native methods class
	Line 3097: 2020-11-19T02:05:38.2853070Z ##[warning]src\oss\IpcFramework\JKang.IpcServiceFramework.Hosting.NamedPipe\NamedPipeNative.cs(118,41): Warning CA2000: Call System.IDisposable.Dispose on object created by 'NamedPipeNative.CreateNamedPipe(' before all references to it are out of scope.**
	Line 3122: 2020-11-19T02:05:39.9612232Z ##[warning]src\oss\IpcFramework\JKang.IpcServiceFramework.TcpTests\HappyPathTest.cs(37,27): Warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
	Line 3125: 2020-11-19T02:05:39.9628006Z ##[warning]src\oss\IpcFramework\JKang.IpcServiceFramework.TcpTests\HappyPathTest.cs(37,44): Warning xUnit1026: Theory method 'HappyPath' on test class 'HappyPathTest' does not use parameter 'input'.
	Line 3128: 2020-11-19T02:05:39.9640386Z ##[warning]src\oss\IpcFramework\JKang.IpcServiceFramework.TcpTests\HappyPathTest.cs(37,58): Warning xUnit1026: Theory method 'HappyPath' on test class 'HappyPathTest' does not use parameter 'expected'.

The named pipe one is the only one that could cause a perf issue. We actually don't use that library, we use Hosting.Tcp, so I'm okay to just ignore them all. WRT to the PowerShell XML comments since this class is only used for PowerShell the XML isnt exposed to the user and the usage is obvious anyway.

**Review by hajya — APPROVED** [review]
thanks for the information  on your process and how you approach this.

**Pull request merged**